src/charmed_kubeflow_chisme/testing/ambient_integration.py

```diff
@@ -0,0 +1,176 @@
+
+    if relate_to_ingress:
+        await model.integrate(
+            f"{ISTIO_INGRESS_K8S_APP}:istio-ingress-route", f"{app}:istio-ingress-route"
+        )
+
+    if relate_to_beacon:
+        await model.integrate(f"{ISTIO_BEACON_K8S_APP}:service-mesh", f"{app}:service-mesh")
+
+    await model.wait_for_idle(
+        raise_on_blocked=False,
+        raise_on_error=True,
+        timeout=900,
+    )
+
+
+async def fetch_response(url: str, headers: dict | None = None) -> tuple[int, str, str]:
+    """Fetch provided URL and return tuple - status, text, and content-type (int, string, string).
+
+    The content-type returned is the media type only (e.g., "text/html"),
+    without charset or other parameters.
+
+    Args:
+        url: The URL to fetch.
+        headers: Optional HTTP headers to include in the request.
+
+    Returns:
+        A tuple of (status_code, response_text, content_type).
+    """
+    async with aiohttp.ClientSession() as session:
+        async with session.get(url, headers=headers) as response:
+            result_status = response.status
+            result_text = await response.text()
+            # Extract only the media type, removing charset and other parameters
+            full_content_type = response.headers.get("Content-Type", "")
+            content_type = full_content_type.split(";")[0].strip()
+    return result_status, result_text, content_type
+
+
+async def assert_path_reachable_through_ingress(
+    http_path: str,
+    namespace: str,
+    service_name: str = "istio-ingress-k8s-istio",
+    headers: dict | None = None,
+    expected_status: int = 200,
+    expected_content_type: str | None = None,
+    expected_response_text: str | None = None,
+) -> None:
+    """Assert that a given path is reachable through the Istio ingress gateway.
+
+    Args:
+        http_path: The HTTP path to test.
+        namespace: The Kubernetes namespace where the Istio ingress gateway is deployed.
+        service_name: The name of the Istio ingress gateway service.
+            Defaults to "istio-ingress-k8s-istio".
+        headers: Optional HTTP headers to include in the request. Defaults to None.
+        expected_status: The expected HTTP status code. Defaults to 200.
+        expected_content_type: Optional content type to check in the response headers.
+            Media type only, e.g., "text/html".
+        expected_response_text: Optional text that should be contained in the response body.
+    """
+    # Get the external IP of the Istio ingress gateway service
+    client = lightkube.Client()
+
+    gateway_svc = client.get(
+        lightkube.resources.core_v1.Service,
+        name=service_name,
+        namespace=namespace,
+    )
+
+    # Check if LoadBalancer ingress is available
+    assert (
+        gateway_svc.status.loadBalancer.ingress
+    ), f"Service {service_name} in namespace {namespace} does not have a LoadBalancer IP"
+
+    external_ip = gateway_svc.status.loadBalancer.ingress[0].ip
```

Comment:
question: is there a case where the should be more than one elements in the ingress list? shall we assert that?